src-backend/api/generator.py

```diff
@@ -54,22 +54,22 @@ def __init__(self, element):
         self.element = element
 
         if not self.element.element_type:
-            self.__raise_error('Element with id %d has no type')
+            self.__raise_error('Error on Page with display index %d: Element has no type')
```

Comment:
include the primary key too:
`Error on Page {primary_key} with display index {display_index}: {reason}`